src/external-interface.ts

```diff
@@ -1,62 +1,46 @@
-interface IBls {
-  Keypair: {
-    generate(): Keypair;
-  };
+export interface IBls {
   PrivateKey: {
-    fromBytes(bytes: Uint8Array): PrivateKey;
-    fromHexString(value: string): PrivateKey;
-    fromInt(num: number): PrivateKey;
-    random(): PrivateKey;
+    fromKeygen(ikm?: Uint8Array): IPrivateKey;
+    fromBytes(bytes: Uint8Array): IPrivateKey;
+    fromHex(hex: string): IPrivateKey;
   };
   PublicKey: {
-    fromPrivateKey(privateKey: PrivateKey): PublicKey;
-    fromBytes(bytes: Uint8Array): PublicKey;
-    fromHex(value: string): PublicKey;
-    fromPublicKeyType(value: IPublicKeyValue): PublicKey;
+    aggregate(pubkeys: IPublicKey[]): IPublicKey;
+    fromBytes(bytes: Uint8Array): IPublicKey;
+    fromHex(hex: string): IPublicKey;
   };
   Signature: {
-    fromCompressedBytes(value: Uint8Array): Signature;
-    fromValue(signature: ISignatureValue): Signature;
-    aggregate(signatures: Signature[]): Signature;
+    aggregate(signatures: ISignature[]): ISignature;
+    fromBytes(bytes: Uint8Array): ISignature;
+    fromHex(hex: string): ISignature;
   };
 
-  initBLS(): Promise<void>;
-  generateKeyPair(): Keypair;
-  generatePublicKey(secretKey: Uint8Array): Buffer;
-  sign(secretKey: Uint8Array, messageHash: Uint8Array): Buffer;
-  aggregateSignatures(signatures: Uint8Array[]): Buffer;
-  aggregatePubkeys(publicKeys: Uint8Array[]): Buffer;
+  sign(secretKey: Uint8Array, messageHash: Uint8Array): Uint8Array;
+  aggregatePubkeys(publicKeys: Uint8Array[]): Uint8Array;
+  aggregateSignatures(signatures: Uint8Array[]): Uint8Array;
   verify(publicKey: Uint8Array, messageHash: Uint8Array, signature: Uint8Array): boolean;
   verifyAggregate(publicKeys: Uint8Array[], messageHash: Uint8Array, signature: Uint8Array): boolean;
-  verifyMultiple(publicKeys: Uint8Array[], messageHashes: Uint8Array[], signature: Uint8Array, fast = false): boolean;
-}
+  verifyMultiple(publicKeys: Uint8Array[], messageHashes: Uint8Array[], signature: Uint8Array): boolean;
 
-interface Keypair {
-  publicKey: PublicKey;
-  privateKey: PrivateKey;
+  init(): Promise<void>;
+  destroy(): void;
 }
 
-interface PrivateKey {
-  getValue(): IPrivateKeyValue;
-  signMessage(message: Uint8Array): Signature;
-  toPublicKey(): PublicKey;
-  toBytes(): Buffer;
-  toHexString(): string;
+export interface IPublicKey {
+  toBytes(): Uint8Array;
+  toHex(): string;
```

Comment:
- missing verify and add/aggregate

Reply:
aggregate is on the constructor. In what situation do you need the `add` method?